src/inmanta/agent/agent.py

```diff
@@ -501,7 +518,34 @@ def get_latest_version_for_agent(self, reason="Unknown"):
                     LOGGER.exception("Failed to receive update")
 
                 if len(resources) > 0:
-                    self._nq.reload(resources, undeployable, reason=reason)
+                    # Set start repair flags correctly
+                    is_normal_deploy_running = self._is_normal_deploy_running()
+                    if (not is_repair_run and self._is_repair_running) or \
+                       (is_repair_run and is_normal_deploy_running):
+                        self._start_repair_when_deployment_finishes = True
+                    elif is_repair_run and self._start_repair_when_deployment_finishes:
+                        self._start_repair_when_deployment_finishes = False
```

Comment:
perhaps add some comments to what this logic intends to achieve and why there is no final else clause.

Reply:
it would also be clearer if you set the condition for `restart or not` in a variable and use that in line 530 instead of splitting the logic over two locations